### PR TITLE
[804] extract privacy page from wizard and make it a single unified standalone page

### PR DESCRIPTION
**Pull request by aldavidson**
### Context

As part of the larger ticket [649 - Support multiple schools per user](https://trello.com/c/lDAgx85e/649-handling-multiple-schools), we'll need to have one `SchoolWelcomeWizard` per user-and-school combination. This means we need to remove the privacy notice from the wizard workflow, and make it standalone (the privacy notice is effectively per user, not per school). A subsequent PR will actually add the school ID to the SchoolWelcomeWizard class and deal with the user journey implications.

In discussion with @fofr we agreed that the privacy notices for school users and responsible body users were the same and could be unified into one - which also means that this work is foundational for supporting multiple RBs per user in future too. 

### Changes proposed in this pull request

* change the default step on `SchoolWelcomeWizard` from 'privacy' to 'allocation'
* add a standalone `PrivacyNoticeController` and move the responsible body privacy notice template to that scope 

There should be no noticeable difference in the user flow.

### Guidance to review

* probably a good idea to check actually clicking through with a new user of all three relevant types (RB, school, hybrid)

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -50,4 +50,27 @@ def render_404_if_feature_flag_inactive(feature_flag)
...
+        school_before_you_can_order_path
+      else
+        school_home_path
+      end
+    elsif user.is_computacenter?
+      computacenter_home_path
+    elsif user.is_support?
+      support_internet_service_performance_path
+    else
+      '/'
```
Comment (fofr) [inline app/controllers/application_controller.rb:73]: Should we log when this happens, given it shouldn't?

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -10,14 +10,6 @@ def create
 
     if @user.save
       token = SessionService.send_magic_link_email!(@user.email_address)
-      # NOTE: this is purely to allow us to display the link in the footer
-      # REMOVE when we have emails actually being sent
-      identifier = @user.sign_in_identifier(token)
-      @debug_info = {
```
Comment (benilovj) [inline app/controllers/users_controller.rb:16]: Shoud we also remove all references to the `show_debug_info` feature flag?

File: db/schema.rb [hunk 3]
```diff
@@ -282,12 +282,13 @@
     t.boolean "is_support", default: false, null: false
     t.boolean "is_computacenter", default: false, null: false
     t.datetime "privacy_notice_seen_at"
-    t.bigint "legacy_school_id"
     t.boolean "orders_devices"
+    t.bigint "legacy_school_id"
```
Comment (benilovj) [inline db/schema.rb:286]: Time to drop this too (maybe a separate PR)

**Review by benilovj — APPROVED** [review]
Seems fine. When testing this, would you be able to just confirm that school users with 'orders_devices: true` are still added to `Computacenter::UserChanges` even after this change? They should, but I'm not sure we have any feature-level specs that confirm this...

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (aldavidson) [reply app/controllers/application_controller.rb:85]: good idea - done

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (aldavidson) [reply app/controllers/users_controller.rb:16]: yes, good idea - but I'll do that as a separate PR

**Comment by aldavidson** [comment]
> Seems fine. When testing this, would you be able to just confirm that school users with 'orders_devices: true`are still added to`Computacenter::UserChanges` even after this change? They should, but I'm not sure we have any feature-level specs that confirm this...

I've added an example to `spec/features/signing_in_as_different_types_of_user_spec.rb` that covers this (and the opposite case)

**Pull request merged**